projects/python-client/src/datapane/client/api/report/core.py

```diff
@@ -265,6 +313,36 @@ def _preprocess_pages(self, pages: t.List[BlockOrPrimitive]):
             # all blocks - wrap as a single page, including layout/mixed/raw elements
             self.pages = [Page(blocks=pages)]
 
+
+class Report(App):
+    """
+    Deprecated (renamed to App)
+    """
+
+    def __init__(
+        self,
+        *arg_blocks: PageOrPrimitive,
+        blocks: t.List[PageOrPrimitive] = None,
+        layout: t.Optional[PageLayout] = None,
+        **kwargs,
+    ):
+        warn(
+            "`Report` is deprecated and will be removed from future releases, use `App` instead",
+            DeprecationWarning,
+        )
+        super().__init__(*arg_blocks, blocks=blocks, layout=layout, **kwargs)
+
+
+class Processor:
```

Comment:
I think we can move `Processor` and all its implementations into a new `processors` module

Reply:
also, let's mark as an abstractclass

Reply:
Sounds good re moving into own module. 

> also, let's mark as an abstractclass

Reports are wrapped in `Processor` to generate report documents for testing, e.g. in `tests.client.local.api.test_reports` we do `dp.Processor(r)._gen_report`

Reply:
Oh ok, yeah that's fine then